### ISSUE TITLE
writable_dirs config

### ISSUE THREAD

**Issue by mehrdaad** [lorisleiva/laravel-deployer#11]
I looked at the docs but could found any.  
Should I specify default values for ``writable_dirs`` when setting it in deploy.php?  
I mean these [ "bootstrap/cache", "storage", "storage/app", "storage/framework", "storage/logs", ... ]

**Comment by mehrdaad** [comment]
Maybe I misconmunicated.  
If I specify this in my deploy.php:  

``add('writable_dirs', ['storage/uploads']);``

Will     ``'bootstrap/cache',
    'storage',
    'storage/app',
    'storage/app/public',
    'storage/framework',
    'storage/framework/cache',
    'storage/framework/sessions',
    'storage/framework/views',
    'storage/logs',`` Still be added to writable_dirs?

**Comment by lorisleiva** [comment]
Yes, if you use the `add` function it will merge your `['storage/uploads']` with the default value.

Note that you can double check all values of your configurations by running:

```bash
php artisan deploy:configs
```

Hope this helps.

**Comment by mehrdaad** [comment]
Just one more question, If I have deployed several versions of my app. now I add a directory to ``shared_dirs``.  
What exactly happens?  
Does previously exising files get copied to one place?

**Comment by lorisleiva** [comment]
* If the new shared dirs are present on your next release it will copy them to the shared folder and symlink them. 
* Otherwise it will create empty dirs in the shared folder and symlink them.

**Comment by mehrdaad** [comment]
directory of storage has mod of 775. which is correct.   But it's user is set as ssh user(root)  instead of webserver.  
Shouldn't it be changed after deployment?

**Comment by lorisleiva** [comment]
If you'd like to configure your writable options further, you can play with the following options: [writable_mode](https://deployer.org/docs/configuration#writable_mode), [writable_use_sudo](https://deployer.org/docs/configuration#writable_use_sudo), [writable_chmod_mode](https://deployer.org/docs/configuration#writable_chmod_mode), [writable_chmod_recursive](https://deployer.org/docs/configuration#writable_chmod_recursive) and [http_user](https://deployer.org/docs/configuration#http_user).

**Comment by mehrdaad** [comment]
Does this deployer change owner,group of writable dirs?

**Comment by lorisleiva** [comment]
Depending on which `writable_mode` you use, yes.
